### PR TITLE
fix(Truncate): Fix truncation markup and makes props more flexible

### PR DESCRIPTION
**Pull request by codecaaron**
### Overview

Fixes an issue where markup wasn't being correctly truncated and makes the API a bit easier to control.  It will also automate use of `Truncate.Atom` which must wrap any React component, so any markup passed to <Truncate> will behave as expected.

* `react-truncate-markup` => 5.1.0
* Small changes to the `Truncate` API to make behavior clearer.
* Alert now always truncates at 2 lines.

### PR Checklist

- [ ] Related to designs:
- [ ] Related to JIRA ticket: ABC-123
- [x] I have run this code to verify it works
- [ ] This PR includes unit tests for the code change

### CODE REVIEW

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-illustrations@0.4.10-alpha.ec6b52.0<br/>@codecademy/gamut-labs@7.13.5-alpha.ec6b52.0<br/>@codecademy/gamut@19.11.6-alpha.ec6b52.0<br/>@codecademy/styleguide@21.20.3-alpha.ec6b52.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://60118193101bb461512ae633--gamut-preview.netlify.app>https://60118193101bb461512ae633--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/60118193101bb461512ae633>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-illustrations@0.4.10-alpha.e8676d.0<br/>@codecademy/gamut-labs@7.14.1-alpha.e8676d.0<br/>@codecademy/gamut@19.11.6-alpha.e8676d.0<br/>@codecademy/styleguide@21.21.1-alpha.e8676d.0<br/></pre>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-illustrations@0.4.10-alpha.a9859a.0<br/>@codecademy/gamut-labs@7.14.1-alpha.a9859a.0<br/>@codecademy/gamut@19.11.6-alpha.a9859a.0<br/>@codecademy/styleguide@21.21.1-alpha.a9859a.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://60119d9857c0b6b182d0249e--gamut-preview.netlify.app>https://60119d9857c0b6b182d0249e--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/60119d9857c0b6b182d0249e>Deploy Logs</a></p>

File: packages/gamut/src/Truncate/index.tsx [hunk 1]
```diff
@@ -1,44 +1,59 @@
...
       ellipsis={<span>...</span>}
       lines={lines}
-      onTruncate={onTruncate}
+      onTruncate={(truncated) => {
+        setIsTruncated(truncated);
+      }}
     >
-      {content}
+      <span>
+        {React.Children.map(children, (child) =>
```
Comment (codecaaron) [inline packages/gamut/src/Truncate/index.tsx:41]: This is required https://github.com/parsable/react-truncate-markup#truncatemarkupatom-

File: packages/gamut/src/Truncate/index.tsx [hunk 2]
```diff
@@ -1,44 +1,59 @@
...
+  /** Truncate markup expects a single child element */
+  const truncatedChildren = (
     <TruncateMarkup
+      tokenize="words"
       ellipsis={<span>...</span>}
       lines={lines}
-      onTruncate={onTruncate}
+      onTruncate={(truncated) => {
+        setIsTruncated(truncated);
+      }}
```
Comment (JoshuaKGoldberg) [inline packages/gamut/src/Truncate/index.tsx:38]: ⛳ ✨ 

```suggestion
      onTruncate={setIsTruncated}
```

File: packages/gamut/src/Truncate/index.tsx [hunk 3]
```diff
@@ -1,44 +1,59 @@
-import cx from 'classnames';
-import React from 'react';
+import React, { isValidElement, useEffect, useState } from 'react';
 import TruncateMarkup from 'react-truncate-markup';
 
-import styles from './styles.module.scss';
+import { Box } from '../Box';
```
Comment (JoshuaKGoldberg) [inline packages/gamut/src/Truncate/index.tsx:4]: 🔥

**Review by JoshuaKGoldberg — APPROVED** [review]
Nice!

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-illustrations@0.4.10-alpha.6a9cb3.0<br/>@codecademy/gamut-labs@7.14.1-alpha.6a9cb3.0<br/>@codecademy/gamut@19.11.6-alpha.6a9cb3.0<br/>@codecademy/styleguide@21.21.1-alpha.6a9cb3.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://6011cb43e426f4ac04036d45--gamut-preview.netlify.app>https://6011cb43e426f4ac04036d45--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/6011cb43e426f4ac04036d45>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-illustrations@0.4.14-alpha.d2f68d.0<br/>@codecademy/gamut-labs@7.16.2-alpha.d2f68d.0<br/>@codecademy/gamut@20.0.1-alpha.d2f68d.0<br/>@codecademy/styleguide@22.0.1-alpha.d2f68d.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://60131368879709963f87439c--gamut-preview.netlify.app>https://60131368879709963f87439c--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/60131368879709963f87439c>Deploy Logs</a></p>

**Pull request merged**